doc/source/docs/reference/sindex.rst

```diff
@@ -0,0 +1,45 @@
+=============
+Spatial index
+=============
+.. currentmodule:: geopandas
+
+GeoPandas offers built-in support of spatial indexing using an R-Tree algorithm. Depending on the ability to import ``pygeos``, GeoPandas will either use ``pygeos.STRtree`` or ``rtree.index.Index``. The main interface for both is the same and follows the ``pygeos`` model. 
+
+``GeoSeries.sindex`` creates a spatial index, which can use the methods and properties documented below.
+
+Constructor
+-----------
+.. autosummary::
+   :toctree: api/
+
+    GeoSeries.sindex
+
+PyGEOS STRtree
+--------------
+.. currentmodule:: geopandas.sindex.PyGEOSSTRTreeIndex
```

Comment:
I understand the concerns. I also initially wanted to have this only once in docs, but then I realised that we document implementation in docstrings. For example in `RTreeIndex.query_bulk`

```
Compatibility layer for pygeos-based sindex.query_bulk.

Iterates over geometry and queries index. This operation is not vectorized and may be slow. Use PyGEOS with query_bulk for speed.
```

Or in pygeos `intersection`:

```
Wrapper for pygeos.query that uses the RTree API.

Compatibility wrapper, use query instead.
```

> Also thinking about how much we should use PyGEOSSTRTreeIndex in the docs (as it will determine the actual url), if in one year it might be called ShapelySTRTreeIndex.

Not ideal, agree. All examples use directly `df.sindex.method` so people will hopefully not call it directly. Once the name is changed, we can add a compatibility wrapper to sindex.py and redirection to docs.

Reply:
Will have a try locally if I can easily combine them, otherwise I agree we can deal with wrappers / redirects later